pkg/vpcmodel/grouping.go

```diff
@@ -332,6 +333,36 @@ func (g *GroupConnLines) groupInternalSrcOrDst(srcGrouping, groupVsi bool) {
 	g.GroupedLines = res
 }
 
+// Go over the output of treating self loops as don't cares - extendGroupingSelfLoops
+// and align all src, dst to use the same reference via g.groupedEndpointsElemsMap
+// this is done here since *GroupConnLines is not known within the extendGroupingSelfLoops context
+func (g *GroupConnLines) unifiedGroupedConnLines(oldConnLines []*GroupedConnLine) []*GroupedConnLine {
+	newGroupedLines := []*GroupedConnLine{}
+	// go over all connections; if src/dst is not external then use groupedEndpointsElemsMap
+	for _, groupedConnLine := range oldConnLines {
+		newGroupedConnLine := &GroupedConnLine{g.unifiedGroupedElems(groupedConnLine.Src),
+			g.unifiedGroupedElems(groupedConnLine.Dst),
+			groupedConnLine.Conn}
+		newGroupedLines = append(newGroupedLines, newGroupedConnLine)
+	}
+	return newGroupedLines
```

Comment:
```suggestion
	newGroupedLines := make([]*GroupedConnLine, len(oldConnLines))
	// go over all connections; if src/dst is not external then use groupedEndpointsElemsMap
	for i, groupedConnLine := range oldConnLines {
		newGroupedLines[i] = &GroupedConnLine{g.unifiedGroupedElems(groupedConnLine.Src),
			g.unifiedGroupedElems(groupedConnLine.Dst),
			groupedConnLine.Conn}
	}
	return newGroupedLines
```

Reply:
I'm ok with the change, but in what aspect is it better? @adisos

Reply:
More efficient to allocate in advance the array, rather than using `append`